### PR TITLE
Update acm stack deployment playbook

### PR DESCRIPTION
**Pull request by NemoDang**

### CODE REVIEW

File: ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml [hunk 1]
```diff
@@ -29,154 +29,163 @@
...
+  when:
+    ENV != 'local'
 
-- name: debug
-  debug:
-    msg: "{{ result.stdout_lines }}"
+- name: Config Alert Manager for local deployment
+  shell: |-
+    sed -i'' --expression 's@<SLACK_API_URL>@'{{ SLACK_API_URL }}'@g' {{ playbook_dir }}/../../cluster-bootstrap/alert-manager-config/overlay/local/app-alertmanager-config.yaml
+    kubectl apply -k ./../../cluster-bootstrap/alert-manager-config/overlay/local
```
Comment (andreadecorte) [inline ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml:189]: can also be replaced with https://docs.ansible.com/ansible/latest/collections/kubernetes/core/kustomize_lookup.html#ansible-collections-kubernetes-core-kustomize-lookup I think, as it is the last kubectl remaining part

File: ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml [hunk 2]
```diff
@@ -29,154 +29,163 @@
...
-
-    printf "=====================Create Openshift Gitops Subscription...\n"
-    kubectl apply -k ./../../cluster-bootstrap/openshift-gitops/deploy
-
-    while [[ $(kubectl get ns openshift-gitops --no-headers --ignore-not-found | wc -l) -eq 0 ]]; 
-        do echo "Waiting for openshift-gitops namespace to be created" && sleep 30;
-    done
+- name: Update argocd configuration
+  template:
+    src: argocd.yaml
```
[the remark this answers is not in the document]
Comment (NemoDang) [reply ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml]: Updated. Thanks.

File: ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml (hunk 1 above)
Comment (NemoDang) [reply ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml]: Update and abstract it to a template.

File: ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml [hunk 3]
```diff
@@ -29,154 +29,163 @@
...
+    state: present
+    src: "{{ item }}"
+  loop:
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/config/argocd.yaml"
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/config/repository.yaml"
+    - "{{ playbook_dir }}/../../cluster-bootstrap/openshift-gitops/config/rolebinding.yaml"
 
+- name: Wait for Openshift GitOps
+  shell: |-
     while [[ $(kubectl get pods -n openshift-gitops --no-headers --ignore-not-found | awk '/openshift-gitops-repo-server/' | wc -l) -ne 1 ]];
```
Comment (andreadecorte) [inline ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml:75]: could you also replace also this with a wait like you did for namespaces? sorry for missing this yesterday

Comment (NemoDang) [reply ansible-playbooks/roles/deploy-acm-stack/tasks/main.yml]: Updated. thx

**Review by andreadecorte — APPROVED** [review]
/approve

**Pull request merged**